src/traces/scattergl/calc.js

```diff
@@ -0,0 +1,38 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var isNumeric = require('fast-isnumeric');
+
+var Axes = require('../../plots/cartesian/axes');
+
+var calcColorscale = require('../scatter/colorscale_calc');
+var arraysToCalcdata = require('../scatter/arrays_to_calcdata');
+
+
+module.exports = function calc(gd, trace) {
+    var xa = Axes.getFromId(gd, trace.xaxis || 'x'),
+        ya = Axes.getFromId(gd, trace.yaxis || 'y');
+
+    var x = xa.makeCalcdata(trace, 'x'),
+        y = ya.makeCalcdata(trace, 'y');
+
+    var serieslen = Math.min(x.length, y.length), i;
+
+    // create the "calculated data" to plot
+    var cd = new Array(serieslen);
+    for(i = 0; i < serieslen; i++) {
+        cd[i] = {x: x[i], y: y[i]};
+    }
```

Comment:
you might even be able to get rid of this loop... it would take a bit of reshaping in `scattergl/select` but can you just say `cd = [{x: x, y: y}]`? Again, if `select` is the only consumer of `cd` you can put whatever you want in there. Regular scatter is much more constrained because it's constructing `cd` as the d3 data for the trace.